### PR TITLE
Fixbug search

### PR DESCRIPTION
**Pull request by gduck**
- category display
- search
- search within categories
- search likes from user page

### CODE REVIEW

File: app/assets/javascripts/angular/controller/home-controller.js.erb [hunk 1]
```diff
@@ -62,38 +37,62 @@ app.controller('HomeCtrl', ['$scope', '$http', '$routeParams', '$location', 'Use
...
+      }      
+    }).error(function(response){
+      console.log(">>>>> ERROR IN searchPINS ", response);
     });
   }
-  $scope.getPins();
+
+  if($routeParams.category) {
+    $http.get('categories.json?keyword=' + $routeParams.category).success(function(response) {
+      $scope.category = response.categories[0];
```
Comment (h8rry) [inline app/assets/javascripts/angular/controller/home-controller.js.erb]: if `categories[0]` will always return 1 item, then it shouldn't be an array

File: app/assets/javascripts/angular/controller/home-controller.js.erb [hunk 2]
```diff
@@ -62,38 +37,62 @@ app.controller('HomeCtrl', ['$scope', '$http', '$routeParams', '$location', 'Use
...
+      $scope.searchPins();
+    })
+    .error(function(response){
+      console.log(">>>>> error in get category");
+    });
+  } else {
+    category_url = '';
+    // if we have no category in the url, get some pins and get more with infinite scroll as you scroll
+    // var end = false;
+    $scope.busy = false;
```
Comment (h8rry) [inline app/assets/javascripts/angular/controller/home-controller.js.erb:74]: @cassiemak 
use a better name for `busy`
use a better name for `after`

File: app/assets/templates/header.html [hunk 3]
```diff
@@ -0,0 +1,61 @@
+<div id="header" class="container-fluid">
```
Comment (h8rry) [inline app/assets/templates/header.html]: This is not new. @cassiemak

File: app/controllers/likes_controller.rb [hunk 4]
```diff
@@ -2,8 +2,13 @@ class LikesController < ApplicationController
 
  def index
     # if current_user
-      @pins = current_user.pins.includes(:category).limit(params[:limit]).offset(params[:offset])
+    if params[:keyword]||params[:category_id]
```
Comment (h8rry) [inline app/controllers/likes_controller.rb]: we need spaces
`params[:keyword] || params[:category_id]`

File: app/assets/javascripts/angular/controller/home-controller.js.erb (hunk 1 above)
Comment (gduck) [inline app/assets/javascripts/angular/controller/home-controller.js.erb]: changed

File: app/controllers/likes_controller.rb (hunk 4 above)
Comment (gduck) [inline app/controllers/likes_controller.rb]: ok

**Pull request merged**